[ROCm] Fix caffe2 build with hipblasv2 api

xw285cornell opened the pull request:
Summary: we need this change along with D52244365 to make caffe2 build happy

Test Plan: OSS CI

Differential Revision: D52275058



cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang

Sabrewarrior commented:
Similar changes are also in #115946

jeffdaily commented on caffe2/utils/math_gpu.cu:
```diff
@@ -854,6 +861,11 @@ CAFFE2_CUDA_EXPORT void GemmBatched<at::Half, CUDAContext>(
+#if defined(USE_ROCM) && ROCM_VERSION >= 60000 && defined(HIPBLAS_V2)
+    auto compute_type = HIPBLAS_COMPUTE_32F;
+#else
+    auto compute_type = HIPBLAS_R_32F;
```
CUDA_R_32F
```suggestion
    auto compute_type = CUDA_R_32F;
```

xw285cornell committed:
[ROCm] Fix caffe2 build with hipblasv2 api (#116073)

Summary:

we need this change along with D52244365 to make caffe2 build happy

Test Plan: OSS CI

Differential Revision: D52275058
